src/storage/v2/vertex_accessor.hpp

```diff
@@ -29,20 +31,35 @@ struct Constraints;
 
 class VertexAccessor final {
  private:
+  struct VertexValidator {
+    explicit VertexValidator(SchemaValidator *schema_validator, const Vertex *vertex)
```

Comment:
Currently, it is not possible due to `VertexAccessot` being copied, and if we have a `VertexValidator` that holds a reference object it cannot be copied, take a look at void `LabelIndex::Iterable::Iterator::AdvanceUntilValid() {` in `indices.cpp`. Do you think its worth doing something about it?

Reply:
I meant only as a constructor parameter, not as a member. So take a reference int the constructor, and then store the address of the object in a member variable. That way I think:
1. it is much more clear for the outside world that they cannot pass into nullptr
2. maybe it also gives a strong hint for the developer that they don't have to deal with nullptr in the member variable.

As I am checking it out, `SchemaValiator` can be passed as `const&`, because all of the interesting member functions are const qualified.

Reply:
Aha ok that makes sense, that's why you only marked the constructor, sorry for the misunderstanding